raft/src/main/java/org/apache/kafka/raft/FileBasedStateStore.java

```diff
@@ -79,56 +74,38 @@ private QuorumStateData loadState(File file, final OffsetAndEpoch logEndOffsetAn
             JsonNode readNode = objectMapper.readTree(line);
 
             if (!(readNode instanceof ObjectNode)) {
-                throw new IllegalStateException("Deserialized node is not an object node");
+                throw new IOException("Deserialized node is not an object node");
             }
             final ObjectNode dataObject = (ObjectNode) readNode;
             final short dataVersion = dataObject.get(DATA_VERSION).shortValue();
```

Comment:
This could raise an NPE. If the version field is not present, we should probably handle it similarly to the `ObjectNode` case above.